File: src/main/java/com/cjcrafter/openai/chat/ChatChoice.java
package com.cjcrafter.openai.chat;

import com.google.gson.JsonObject;

public final class ChatChoice {

    private final int index;
    private final ChatMessage message;
    private final String finishReason;

    /**
     * Holds the data for 1 generated text completion.
     *
     * @param index        The index in the array... 0 if n=1.
     * @param message      The generated text.
     * @param finishReason Why did the bot stop generating tokens?
     */
    public ChatChoice(int index, ChatMessage message, String finishReason) {
        super();
        this.index = index;
        this.message = message;
        this.finishReason = finishReason;
    }

    public ChatChoice(JsonObject json) {
        this(json.get("index").getAsInt(), new ChatMessage(json.get("message").getAsJsonObject()), json.get("finish_reason").toString());
    }

    public int getIndex() {
        return this.index;
    }

    public ChatMessage getMessage() {
        return this.message;
    }

    public String getFinishReason() {
        return this.finishReason;
    }
}

File: src/main/kotlin/com/cjcrafter/openai/chat/ChatChoice.kt
package com.cjcrafter.openai.chat

import com.cjcrafter.openai.FinishReason
import com.google.gson.JsonObject
import com.google.gson.annotations.SerializedName

/**
 * The OpenAI API returns a list of `ChatChoice`. Each choice has a
 * generated message ([ChatChoice.message]) and a finish reason
 * ([ChatChoice.finishReason]). For most use cases, you only need the generated
 * message.
 *
 * By default, only 1 [ChatChoice] is generated (since [ChatRequest.n] == 1).
 * When you increase `n`, more options are generated. The more options you
 * generate, the more tokens you use. In general, it is best to **ONLY**
 * generate 1 response, and to let the user regenerate the response.
 *
 * @property index        The index in the array... 0 if [ChatRequest.n]=1.
 * @property message      The generated text.
 * @property finishReason The reason the bot stopped generating tokens.
 * @constructor Create a new chat choice, for internal usage.
 * @see FinishReason
 */
data class ChatChoice(
    val index: Int,
    val message: ChatMessage,
    @field:SerializedName("finish_reason") val finishReason: FinishReason
)